Automattic/jetpack
# Form Block: add options to install/activate Jetpack CRM

Fixes #21439

#### Changes proposed in this Pull Request:

* This is a first pass at this. Ideally, I think we should extract some of this functionality into the shared-extension-utils, alongside the rest of the plugin-management utils, so it could be used by other blocks, and would avoid duplicating code.

#### Jetpack product discussion

* N/A

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Start from a site with Jetpack installed with this branch, and with the Contact form module active. A JN site can be perfect for that, launched using [Jetpack Live Branches](https://github.com/Automattic/jetpack/blob/master/tools/jetpack-live-branches/README.md).
* Also install the WP Rollback plugin, we'll need it later.
* Go to Pages > Add New, and add a form block.
* In the form settings sidebar (ensure you're selecting the form block, and not a child block of the form block), you will see option panels, including a CRM panel.
* From there, try to install the CRM plugin.
* Head over to the Plugins menu, and deactivate the plugin.
* Go back to the page editor, and try to activate the plugin.
* Play with the toggle to save form entries to the CRM. It should get saved properly when saving the page.
* Go to the Plugins menu again, and use the Rollback option to go back to an old version of the plugin, before version 3.0
* Go back to the page editor, and notice that you now have a prompt to update the plugin instead of the toggle.
